### PR TITLE
Partially implemented advanced data table

### PR DESCRIPTION
**Pull request by ferspanghero**

### CODE REVIEW

File: turk-one/app/Views-Controllers/Admin/admin-datatable.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		if (arrayOfConditionals.length > 0) {
+			return {$or: arrayOfConditionals};
+		} else {
+			return {}
+		}
+	}
+
+	Template.adminDataTable.helpers({
+		workerTickets: function () {
+			// TODO: Store session keys someplace so we don't have to remember the keys specifically each time
```
Comment (sahandnayebaziz) [inline turk-one/app/Views-Controllers/Admin/admin-datatable.js:16]: Yes! Session keys not being strings is something I would **love**. In Swift, I would implement these as enums to make them less stringy and more enforced... Wonder what standard practice is for doing this in Javascript.

File: turk-one/app/Views-Controllers/Admin/admin-datatable.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+				}
+			});
+
+			return moment.duration(totalSolutionTime, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		infoModalTimeFormatted: function () {
+			return moment.duration(this.infoModalTime, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		averageSolutionTime: function () {
+			var solutions = Solutions.find({workerId: this._id}, {sort: {canvasNumber: 1}});
```
Comment (sahandnayebaziz) [inline turk-one/app/Views-Controllers/Admin/admin-datatable.js:139]: You may need to add .fetch() to the end of this Solutions.find({}), no? .find is asynchronous and may not be returned by the time the code moves on. Adding .fetch() to the end says, to my understanding, "make this a synchronous array". If this is already working, then great :) If it sometimes misses, try .fetch() at the end!

File: turk-one/app/Views-Controllers/Admin/admin-datatable.html [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                                <th>Solution 5 - ID</th>
+                                <th>Solution 5 - Sketch time</th>
+                                <th>Solution 5 - Sketch complexity</th>
+                                <th>Solution 5 - Description length</th>
+                                <th>Solution 5 - Description</th>
+                            </tr>
+                            </thead>
+                            <tbody>
+                            {{#each workerTickets}}
+                                {{> workerDataTableRow}}
```
Comment (sahandnayebaziz) [inline turk-one/app/Views-Controllers/Admin/admin-datatable.html:58]: Perfect. Great job picking up on this

**Comment by sahandnayebaziz** [comment]
Looks great to me. Happy to merge this in because it already looks like it will be helpful for Edgar's work. Good job

**Comment by egster** [comment]
Nice work fernando. 

Small update:
I made an additional branch for the quality review page. Is deployed on the server right now for the reviewing on Thursday.

Al the solutions are printed and ready for tomorrow's categorization. 

Modulus now has a third db with both db's from both experiments merged. This database still needs some cleaning to make it a but faster. Solutions has over 3000 items now.

File: turk-one/app/Views-Controllers/Admin/admin-datatable.js (hunk 2 above)
Comment (ferspanghero) [inline turk-one/app/Views-Controllers/Admin/admin-datatable.js:139]: Thanks for the tip, thought it was synchronous. Fixed.

Comment (ferspanghero) [inline turk-one/app/Views-Controllers/Admin/admin-datatable.js:139]: Correction: I made the change but it stopped loading correctly. I really don't want to spend time understanding this, but I couldn't find easily anywhere saying that .find is async.

Rather, the difference between .find and .fetch seems more related to how data is rendered: http://stackoverflow.com/questions/16601957/meteor-use-fetch-or-find-in-template-helper-functions.

I'll change it back to .find for now.

**Pull request merged**